st3/lsp_utils/npm_client_handler.py

```diff
@@ -17,39 +36,53 @@
 }  # type: ignore
 
 
-class ApiWrapper(object):
+class ApiWrapper:
+
     def __init__(self, client):
         self.__client = client
 
     def on_notification(self, method: str, handler: Callable) -> None:
-        self.__client.on_notification(method, handler)
+        if USE_NEW_API:
+            setattr(self.__client, method2attr(method), handler)
+        else:
+            self.__client.on_notification(method, handler)
 
     def on_request(self, method: str, handler: Callable) -> None:
-        def on_response(params, request_id):
-            handler(params, lambda result: send_response(request_id, result))
+        if USE_NEW_API:
+
+            def handler_wrapper(this, params, request_id):
+                handler(params, lambda result: this.send_response(Response(request_id, result)))
 
-        def send_response(request_id, result):
-            self.__client.send_response(Response(request_id, result))
+            setattr(self.__client, method2attr(method), handler_wrapper)
```

Comment:
Definitively would be good to have a method for registering notification/request handlers.